### PR TITLE
Simplify `LoadServicesFromModulesPass::process`

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Simplify `LoadServicesFromModulesPass::process`.
| Type?             | refacto
| Category?         | CO 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A.
| How to test?      | Make sure this PR work fine https://github.com/PrestaShop/PrestaShop/pull/27969.
| Possible impacts? | Potential regression here https://github.com/PrestaShop/PrestaShop/issues/27968 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/28088)

### CODE REVIEW

File: src/PrestaShopBundle/DependencyInjection/Compiler/LoadServicesFromModulesPass.php [hunk 1]
```diff
@@ -54,39 +59,29 @@ class LoadServicesFromModulesPass implements CompilerPassInterface
...
         $this->configPath = '/config/' . (empty($containerName) ? '' : trim($containerName, '/') . '/');
+        $this->activeModulesPaths = (new ModuleRepository())->getActiveModulesPaths();
     }
 
     /**
      * {@inheritdoc}
      */
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasParameter('kernel.active_modules')) {
```
Comment (matks) [inline src/PrestaShopBundle/DependencyInjection/Compiler/LoadServicesFromModulesPass.php:64]: Hi @PululuK 

Current way of retrieving active modules is using this parameter `kernel.active_modules`

ModuleRepository:: getActiveModulesPaths() logic seem different. It's based on the database content. So isn't this changing the behavior?


https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Module/Repository/ModuleRepository.php

Comment (PululuK) [reply src/PrestaShopBundle/DependencyInjection/Compiler/LoadServicesFromModulesPass.php:64]: @matks see please https://github.com/PrestaShop/PrestaShop/blob/develop/app/AppKernel.php#L192

Comment (matks) [reply src/PrestaShopBundle/DependencyInjection/Compiler/LoadServicesFromModulesPass.php:64]: Ha 😄 

But then isn't this PR duplicating the behavior? 🤔

Comment (PululuK) [reply src/PrestaShopBundle/DependencyInjection/Compiler/LoadServicesFromModulesPass.php:64]: Can you please read the full code ?

Comment (PululuK) [reply src/PrestaShopBundle/DependencyInjection/Compiler/LoadServicesFromModulesPass.php:64]: @matks  currently : 
- You use `getModulesPaths` and loop to check if exists in modules active


`ModuleRepository::getActiveModulesPaths` is equivalent, so more appropriate here I think.

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]
looks good to me, @matks do you want to check again before merging?

**Review by matks — APPROVED** [review]
OK for me

@PululuK thank you for explanations

**Review by AureRita — APPROVED** [review]
Hi @PululuK 

Thank you for your PR, I tested it and it semms to works well 

as you can see here 

https://user-images.githubusercontent.com/100699445/161990441-16acf13f-ec46-41c5-921d-f59773196943.mp4

I tested it on Multishop, 
the Enable/disabled/Reset/Install/Uninstall
with QA module

**Comment by Progi1984** [comment]
Thanks @PululuK & @AureRita

**Pull request merged**